plateau.ts

```diff
@@ -0,0 +1,16 @@
+export default class Plateau {
```

Comment:
At some places we have followed function based implementation, and at some places, class based approach. Let's be consistent in what approach we'd like to take. Anyone is good.
Also, might be a good idea to read about which is more efficient and better suited for design patterns.

Reply:
I personally like OOP based approach, but I could be wrong

Reply:
Need to think on this.

The way I see it or designed is that we create class where objects are to be created from it. So, for plateau and rover it makes sense. But keeping validate() and explore() logic in one of these classes of course wouldn't look clean. I can group validate and explore together into another class but question I have is that is it necessary to always group function into classes ?

validate() and explore() can be standalone functions having a single purpose. So, I'd call them utility or helper functions.

Reply:
There's no harm in having mixed implementation. But then there has to be a clear boundary between what is supposed to be created as functional vs what is created as class.
In your approach, the domain logic is split into functional + class components, which makes it harder to follow design patterns too. You can always keep utility functions like, say "parseInput" as functions, but `explore`speciially  looks more like a function of the rover than a utility function.

Reply:
Yes, you're right. I've thought of keeping explore as a method of rover but it'd simply make our class a bit sophisticated or complex looking. I wanted all components to be minimal and clean. With move() method we're abstracting logic of explore().

Anyway, I'm checking on how to fit explore function such that it makes sense.

Reply:
I would suggest to think through the approach first and then implement. The goal is not to implement a class or a function, the goal is to make a readable, thought-through solution.